File: ecmascript/bob/bob.js
class Bob {
  hey(message) {
    let question = (m) => m.slice(-1) === '?' && !yell(m);
    let yell = (m) => m.toUpperCase() === m && /[A-Z]/.test(m);
    let silence = (m) => m.trim() === '';

    switch (true) {
      case question(message):
        return 'Sure.';
      case yell(message):
        return 'Whoa, chill out!';
      case silence(message):
        return 'Fine. Be that way!';
      default:
        return 'Whatever.';
    }
  }
}

export default Bob;

File: elixir/bob/bob.exs
defmodule Message do
  @moduledoc """
    Defines functions that determinate sentence type, like a question, shouting and silence.
  """
  def is_question(input), do: String.ends_with?(input, "?")
  def is_yell(input), do: String.upcase(input) == input && String.downcase(input) != input
  def is_silence(input), do: String.trim(input) == ""
end

defmodule Bob do
  import Message

  def hey(input) do
    cond do
      is_question(input) -> "Sure."
      is_yell(input) -> "Whoa, chill out!"
      is_silence(input) -> "Fine. Be that way!"
      true -> "Whatever."
    end
  end
end
